### PR TITLE
Added document title to UserLayout

### PR DESCRIPTION
**Pull request by reggiegutter**
Added document title to UserLayout by identifying the current route object and using its name kaey to set the title

### CODE REVIEW

File: src/layouts/UserLayout.js [hunk 1]
```diff
@@ -31,26 +32,51 @@ const copyright = (
...
-    </div>
-    <div className={styles.content}>
-      <div className={styles.top}>
-        <div className={styles.header}>
-          <Link to="/">
-            <img alt="logo" className={styles.logo} src={logo} />
-            <span className={styles.title}>Ant Design</span>
-          </Link>
+class UserLayout extends Component {
+  getPageTitle() {
```
Comment (chenshuai2144) [inline src/layouts/UserLayout.js]: 这个方法可以提出去，传入一个 props 就好了

File: src/layouts/UserLayout.js [hunk 2]
```diff
@@ -31,26 +32,51 @@ const copyright = (
...
+      location: { pathname },
+      route: { routes },
+    } = this.props;
+
+    const currentRoute = routes.filter(route => route.path === pathname);
+
+    let title = 'Ant Design';
+    if (currentRoute.length > 0) {
+      if (currentRoute[0].name) {
+        title = `${currentRoute[0].name} - Ant Design`;
```
Comment (chenshuai2144) [inline src/layouts/UserLayout.js]: currentRoute.length > 0 &&   currentRoute[0] &&  currentRoute[0].name

**Comment by chenshuai2144** [comment]
整个国际化怎么样，就像别的 menu 一样

File: src/layouts/UserLayout.js (hunk 2 above)
Comment (afc163) [reply src/layouts/UserLayout.js]: `currentRoute[0]?.name` ？

Comment (chenshuai2144) [reply src/layouts/UserLayout.js]: currentRoute 是个数组，取得数组的第一个。

File: src/layouts/UserLayout.js (hunk 1 above)
Comment (xiaohuoni) [inline src/layouts/UserLayout.js]:
```js
 getPageTitle = (pathname, breadcrumbNameMap) => {
    const currRouterData = this.matchParamsPath(pathname, breadcrumbNameMap);

    if (!currRouterData) {
      return title;
    }
    const pageName = menu.disableLocal
      ? currRouterData.name
      : formatMessage({
          id: currRouterData.locale || currRouterData.name,
          defaultMessage: currRouterData.name,
        });

    return `${pageName} - ${title}`;
  };
```

**Comment by chenshuai2144** [comment]
https://github.com/ant-design/ant-design-pro/pull/3564

**Pull request closed without merging**